### PR TITLE
Load diagnosis on patient index page

### PR DESCRIPTION
**Pull request by mariadenadel**
image

This Pull Request adds a new button "Load diagnosis items". When you click on this button this will show you all diagnosis items for a particular patient. If there are no diagnosis items for a patient a special message will appear.

### CODE REVIEW

File: rails-homework/app/views/patients/index.html.erb [hunk 1]
```diff
@@ -30,13 +30,19 @@
...
-          <%= link_to "Delete", patient_path(patient.id), method: :delete, data: { confirm: "Are you sure?" }, class: "btn btn-outline-danger" %>
-      <% end %>
+      <td class="actions">
+        <% if can? :update, @patients %>
+            <%= link_to "Edit", edit_patient_path(patient.id), class: "btn btn-outline-primary" %>
+        <% end %>
+        <% if can? :destroy, @patients %>
+            <%= link_to "Delete", patient_path(patient.id), method: :delete, data: { confirm: "Are you sure?" }, class: "btn btn-outline-danger" %>
+        <% end %>
+        <% if can? :manage, @diagnosis_items %>
```
Comment (olehvavryniv) [inline rails-homework/app/views/patients/index.html.erb]: Can view?

Comment (mariadenadel) [reply rails-homework/app/views/patients/index.html.erb]: `<% if can? :manage, @diagnosis_items %>` wrong?
I think `<% if can? :read, @diagnosis_items %>` will be better

Comment (olehvavryniv) [reply rails-homework/app/views/patients/index.html.erb]: Yes

File: rails-homework/app/javascript/packs/load_diagnosis_items.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  $(".load-diagnosis").on("click", function(event){
+    let patientId = event.target.id
+    if ($(`#patient_id${patientId}`).children().length === 0) {
+      $.getJSON(
+        `/diagnosis_items.json?patient_id=${patientId}`,
+        function(data) {
+          let diagnosisItems = []
+          for (let key in data){
+            diagnosisItems.push(data[key].content);
+          }
```
Comment (olehvavryniv) [inline rails-homework/app/javascript/packs/load_diagnosis_items.js]:
```suggestion
          const diagnosisItems = data.map((item) => item.content)
```

File: rails-homework/app/javascript/packs/diagnosis_items_loader.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    diagnosisItems.forEach(function(item, index){
+      output += `<div class="card card-body border-light" id="patient${currentPatientId}_${index}">${item}</div>`
+    })
+  } else {
+    output = "<div class='card card-body text-secondary'>There is no diagnosis</div>"
+  }
+  containerForDiagnosisItems.append(output)
+}
+
+function handler(event){
```
Comment (olehvavryniv) [inline rails-homework/app/javascript/packs/diagnosis_items_loader.js:14]: You need better name here. Something like diagnosisClickHandler

**Review by olehvavryniv — APPROVED** [review]

**Review by olehvavryniv — APPROVED** [review]

**Pull request merged**